generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/mapping/MediaTypeToSchemaIdGenerator.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.services.generators.commons.mapping;
+
+import static uk.gov.justice.services.generators.commons.helper.GeneratedClassWriter.writeClass;
+import static uk.gov.justice.services.generators.commons.helper.Names.MAPPER_PACKAGE_NAME;
+import static uk.gov.justice.services.generators.commons.helper.Names.packageNameOf;
+
+import uk.gov.justice.raml.core.GeneratorConfig;
+
+import com.squareup.javapoet.TypeSpec;
+import org.raml.model.Raml;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Generates MediaType to schema id mapping class for a given RAML object
+ */
+public class MediaTypeToSchemaIdGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MediaTypeToSchemaIdGenerator.class);
+    private final SchemaMappingClassNameGenerator schemaMappingClassNameGenerator = new SchemaMappingClassNameGenerator();
+    private final MediaTypeToSchemaIdParser mediaTypeToSchemaIdParser = new MediaTypeToSchemaIdParser(new SchemaIdParser());
+    private final MediaTypeToSchemaIdMapperClassBuilder mediaTypeToSchemaIdMapperClassBuilder = new MediaTypeToSchemaIdMapperClassBuilder(schemaMappingClassNameGenerator);
+
```

Comment:
Can we use injection here - rather than instantiating everything in the class? Or is this another example of were we can't

Reply:
This is created from the generator plugin...no access to CDI from here